sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/ClientRetryPolicy.java

```diff
@@ -161,6 +164,20 @@ public Mono<ShouldRetryResult> shouldRetry(Exception e) {
                 clientException);
         }
 
+        if (clientException != null && Exceptions.isStatusCode(clientException, HttpConstants.StatusCodes.REQUEST_TIMEOUT)) {
+            logger.info(
```

Comment:
Does this has to be info or is debug sufficient?

Reply:
It is a terminal exception which could possibly lead to exception bookmarking for circuit breaker - might help to diagnose in the future why a region was quarantined.

Reply:
I am on the side feel it can be chatty - currently we have a similar log statement for service unavailable, and as part of it, the cosmosDiagnostics also get logged, which may flood the logging system.

Can we encapsulate some of the intention in the diagnostics/ for example maybe also logging how many requests failed in a region

Reply:
I can capture failure/success statistics in the diagnostics and make the log `debug` level.